firmware/library/L1_Drivers/ssp.hpp

```diff
@@ -0,0 +1,206 @@
+/**
+ ** Provide How-to for driver
+
+ **
+ **
+ */
+
+#pragma once
+
+#include "LPC40xx.h"
+#include "pin_configure.hpp"
+
+class SSPInterface
+{
+ public:
+    // SSP peripherals
+    enum Peripheral
+    {
+        SSP0 = 0,
+        SSP1 = 1,
+        SSP2 = 2
+    };
+
+    enum PowerON
+    {
+        PCONP0 = 21,
+        PCONP1 = 10,
+        PCONP2 = 20
+    };
+
+    // SSP frame formats
+    enum FrameModes
+    {
+        SPI     = 0,
+        TI      = 1,
+        MICRO   = 2
+    };
+
+    // SSP Master/slave modes
+    enum MSMode
+    {
+        MASTER  = 0;
+        SLAVE   = 1
+    };
+
+
+    // SSP data size for frame packets
+    enum DataSize

```

Comment:
For these to actually be binary, you need to prefix them with `0b`. For example FOUR should be `0b0011`. Otherwise your number is eleven.

Reply:
Since DataSize is not used here, it shouldn't be in the interface. But I am wondering why there isn't a method in the interface that uses this enum. If looks like you are missing quite a few methods for your interface.

Reply:
As far as the enums go, they are all being used by the struct in the implementation which in turn is used by init(). I think I may have to move all the enums to the implementation. I'm also planning on adding getter and setter functions, which I'm assuming should be added to the interface as well?

Reply:
Yes, define them in the implementation class, but not here if they aren't used here. But, if you are adding setters and getters that use these, then that is fine to keep them there.